staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -255,10 +257,16 @@ func (b *Builder) FilenameParam(enforceNamespace bool, filenameOptions *Filename
 			}
 			b.URL(defaultHttpGetAttempts, url)
 		default:
-			if !recursive {
+			matches, err := expandIfFilePattern(s)
```

Comment:
I talked to @liggitt and we should do an initial `os.Stat` on the original string just in case someone absurdly named their file with a `*` in it.

Reply:
it looks like Glob also handles characters like `{}[]?\` specially... those are more likely to show up in actual filenames in ways Glob would consider invalid

for example, `test[].yaml` is a valid filename that calling Glob would break on

Reply:
Shouldn't we do both? Return exact matches (if it exists) along with glob results?

Reply:
No, if they specify an exact match that is an actual file or dir, that should take precedence

Reply:
All set. If the input is an invalid pattern or if the file name contains a pattern itself then the raw input is used. Otherwise expanded.